src/main/scala/com/evolutiongaming/crypto/ConfigDecrypter.scala

```diff
@@ -0,0 +1,79 @@
+package com.evolutiongaming.crypto
+
+import com.typesafe.config.{Config, ConfigFactory}
+
+import scala.util.Try
+
+/**
+  * Provides decryption of passwords encrypted using [[Crypto.encryptAES]] for Typesafe Config.
+  *
+  * To use extension methods for [[Config]], import from [[ConfigDecrypter.syntax]].
+  *
+  * Configuration parameters:
+  * - `encryptedPasswords` - if not set or `false` password values are treated as plain-text,
+  * if `true` - as cipher-text
+  * - `application.secret` - private key string to use in decryption
+  */
+trait ConfigDecrypter {
+  /**
+    * If set password values are expected to be encrypted, plain-text otherwise
+    */
+  def encryptedPasswordsEnabled: Boolean
+  /**
+    * Decrypts a password value obtaining the plain text
+    */
+  def decryptString(encryptedPassword: String): Try[String]
+  /**
+    * Decrypts a password value obtaining the plain text - throws exceptions if the plain text recovery is not
+    * possible
+    */
+  def decryptStringUnsafe(encryptedPassword: String): String
+  /**
+    * Decrypts a password value at a config path obtaining the plain text
+    */
+  def decryptPath(configPath: String): Try[String]
+  /**
+    * Decrypts a password value at a config path obtaining the plain text - throws exceptions if
+    * the plain text recovery is not possible
+    */
+  def decryptPathUnsafe(configPath: String): String
```

Comment:
This one doesn't really map to API of Config interface.
`config.getString("username")`
`config.decryptPathUnsafe("password")`

In one project I used `def getProtectedString(path: String)` with encryption key as implicit, but it was situational. Though, the name better matches the config API.
`config.getString("username")`
`config.getProtectedString("password")`
You don't need to use `Unsafe` prefix for this, as the original `getString` throws when path is not found and we are using unchecked exceptions. For variant with `Try` result you can then use `Safe` suffix.

Also. I would split it into 2 traits. One with `decryptString` and one for working with Config paths and mode switch (will comment about it in implementation part).

It is not an issue on its own, just an opinion on the design.

Reply:
I didn't want to change the original semantics too much. As I understand it, given that in some environments you have encrypted password and in others - plain, all the code accessing config secrets would have to do that `encryptedPasswordsEnabled` check one way or another, even for `decryptString`. At least that's what we do in our service.

I don't like the fact that `decrypt` doesn't do actual decryption sometimes though, it just feels wrong as a method naming.

BTW I like the idea of sticking to the Config API style, I'll think what I can make out of it.

Reply:
Yeah, if decryptString was used this way in real code, then it is better to keep this as before (I haven't checked how popular this one was, expected that decryptPath had way more if not all usages). An option would be to provide the new methods with Config API style and deprecate the old ones in future releases.

Reply:
On internal gitlab:
- decryptString - 12 usages
- decryptPath - 1 usage

:-(